### ISSUE TITLE
"Debug Failure. False expression:" when trying to augment imported modules on 2.1.4

### ISSUE THREAD

**Issue by StrikeForceZero** [Microsoft/TypeScript#13303]
**TypeScript Version:**  2.1.4

**Code**

```json
{
  "name": "typescript-error-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@types/node": "^6.0.57",
    "chalk": "^1.1.3",
    "typescript": "^2.1.4"
  }
}

```

tsconfig.json
```json
{
  "compilerOptions": {
    "module": "commonjs",
    "target": "es6",
    "inlineSourceMap": true,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "preserveConstEnums": true,
    "lib": [
      "es5",
      "es6",
      "es7",
      "es2017.object"
    ],
    "typeRoots": [
      "node_modules/@types"
    ],
    "outDir": "build"
  },
  "exclude": [
    "build",
    "node_modules"
  ]
}
```

example.ts
```ts
// chalk has @types/chalk for definitions 
// but for the sake of brevity I used it as a quick example pretending it had no definitions
import foo = require('chalk');
import FooBar = foo.Foo.Bar;
console.log('foo');
```

types.d.ts
```ts
export namespace Foo {
    interface Bar {

    }
}

declare module "chalk" {
    export = Foo;
}
```

running `./node_modules/.bin/tsc -P .` gets the error

Interestingly, If you remove the following it will compile:
```ts
declare module "chalk" {
    export = Foo;
}
```

**Expected behavior:**

No error.

**Actual behavior:**

```shell
node_modules\typescript\lib\tsc.js:1794
                throw new Error("Debug Failure. False expression: " + (message || "") + verboseDebugString);
                ^

Error: Debug Failure. False expression:
    at Object.assert (\node_modules\typescript\lib\tsc.js:1794:23)
    at resolveExternalModule (\node_modules\typescript\lib\tsc.js:20568:30)
    at resolveExternalModuleNameWorker (\node_modules\typescript\lib\tsc.js:20535:20)
    at mergeModuleAugmentation (\node_modules\typescript\lib\tsc.js:19892:34)
    at initializeTypeChecker (\node_modules\typescript\lib\tsc.js:35233:25)
    at Object.createTypeChecker (\node_modules\typescript\lib\tsc.js:19739:9)
    at getDiagnosticsProducingTypeChecker (\node_modules\typescript\lib\tsc.js:49608:93)
    at Object.getGlobalDiagnostics (\node_modules\typescript\lib\tsc.js:49885:41)
    at compileProgram (\node_modules\typescript\lib\tsc.js:51966:78)
    at compile (\node_modules\typescript\lib\tsc.js:51925:26)
```

**Comment by mhegazy** [comment]
and i am not sure i understand what you were trying to do.. but if you want to declare module `chalk`, make sure types.d.ts is a global and not a module (i.e. remove the top level `export` in `export namespace Foo`, that turns the file into a module and it has its own scope now). if you want to augment it, then you need to have it declared first. so you can install `@types/chalk` for that.

**Comment by StrikeForceZero** [comment]
@mhegazy whoops, top level export was not intentional. What I'm trying to do is create a type definition for an untyped module (chalk was a poor example as there is a type definition available for it `@types/chalk`). The goal was, anywhere that imports this untyped module, now has this drop in definition applied to the import.

I actually was testing this in typescript@next prior to opening an issue but didn't understand why declaring definitions for untyped modules became unsupported. Searching issues I noticed a few issues potentially related, but nothing explaining the new error in my case. Figured I'd open up an issue with a clear title targeting typescript@latest.

Just now, I realized the possibility of untyped module definitions now being required to be inside a folder listed under the `tsconfig.json` `compilerOptions.typeRoots` property. 

Testing this by moving the `types.d.ts` in my example to a folder called `types` and adding `"types"` to typeRoots in my tsconfig.json example seemed to do trick.

Did I overlook this in the documentation? I feel like the error message could say something equivalent to: "Module 'foo' resolves to an untyped module at './node_modules/foo/index.js', which cannot be augmented. Did you forget to specify a typeRoots directory containing this definition?".
